examples/standalone_mode_demo.ipynb

```diff
@@ -0,0 +1,609 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Graphstorm Standalone Mode Demonstration\n",
+    "\n",
+    "In this notebook, we'll demonstrate how to use the standalone mode of Graphstorm. The standalone mode is primarily designed for model developers to prototype their data and model training/evaluation/inference pipelines using a single machine. \n",
+    "\n",
+    "---\n",
+    "\n",
+    "## Setup \n",
+    "\n",
+    "This notebook requires installing graphstorm using pip. Please find [more details on installation of graphstorm](https://graphstorm.readthedocs.io/en/latest/install/env-setup.html#setup-graphstorm-with-pip-packages)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# !pip install graphstorm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from argparse import Namespace\n",
+    "import matplotlib.pyplot as plt\n",
+    "# import graphstorm related stuffs\n",
+    "import graphstorm as gs\n",
+    "from graphstorm.config import GSConfig\n",
+    "from graphstorm.trainer import GSgnnNodePredictionTrainer\n",
+    "from graphstorm.dataloading import GSgnnNodeTrainData, GSgnnNodeDataLoader, GSgnnNodeInferData\n",
+    "from graphstorm.inference import GSgnnNodePredictionInferrer\n",
+    "from graphstorm.eval import GSgnnAccEvaluator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# init the graphstorm \"cluster\"\n",
+    "def init_standalone_environment():\n",
+    "    \"\"\"Initialize standalone environment for graphstorm\"\"\"\n",
+    "    env_vars = {\n",
+    "        'RANK': '0',\n",
+    "        'WORLD_SIZE': '1',\n",
+    "        'MASTER_ADDR': '127.0.0.1',\n",
+    "        'MASTER_PORT': '54321'\n",
+    "    }\n",
+    "    # set up required ENV_VAR for distributed context\n",
+    "    for key, val in env_vars.items():\n",
+    "        os.environ[key] = val\n",
+    "\n",
+    "    gs.initialize(ip_config=env_vars['MASTER_ADDR'], backend='gloo')\n",
```

Comment:
`env_vars` is not necessary. you can set `ip_conifg=None`.

Reply:
We could do this, but it will leads to this bug: https://github.com/awslabs/graphstorm/issues/535
Will switch to the following after bug fix.